lib/active_model/serializer.rb

```diff
@@ -102,7 +111,16 @@ def self.adapter
     end
 
     def self.root_name
-      name.demodulize.underscore.sub(/_serializer$/, '') if name
+      name.demodulize.underscore.sub(/_serialization$/, '') if name
+    end
+
+    def self.deserialize(params)
+      association_keys = _reflections.map(&:name)
+      permitted_params = adapter.params(@_params, association_keys) || @_params
+      permitted_key    = adapter.root || root_name
+
+      sanitazed_params = params.require(permitted_key.to_sym).permit(permitted_params)
```

Comment:
so, a couple things, how I feel about using strong_params

1. I personally never really got comfortable with strong params
1. AMS will start raising unpermitted parameter exceptions and such, which might be unexpected? The serializer, IIRC, ignores extra info, which I think is being 'liberal in what you accept'.
1. deserialize will no longer take in a hash, but an instance of [`ActionController:: Parameters`](https://github.com/rails/strong_parameters#use-outside-of-controllers) (which I suppose we could do that transformation here...)
1. it would [add a dependency on action_controller/action_pack which AMS currently does not have](https://github.com/rails/strong_parameters/blob/42397cc023b78a671580e35ba6013b6f4a8fe54a/strong_parameters.gemspec)
1. You could just `params.fetch(root) { fail KeyError, "Missing root in #{params.inspect}" }.slice(permitted_params)` (after requiring [Hash#slice](https://github.com/rails/rails/blob/4-2-stable/activesupport/lib/active_support/core_ext/hash/slice.rb))
```